### PR TITLE
Swift penitence conduit

### PR DESCRIPTION
**Pull request by oisinod**
added support for swift penitence conduit, has atonement healing, direct healing and dps

### CODE REVIEW

File: .env.local.example [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# Copy this file and rename it to `.env.local` to configure your own WCL API key. You can get your own API key from the bottom of the Warcraft Logs settings: https://www.warcraftlogs.com/profile
-# You should also enter the application name "WoWAnalyzer (development)" on that page, just above the key.
-REACT_APP_WCL_API_KEY=INSERT_YOUR_OWN_API_KEY_HERE
```
Comment (emallson) [inline .env.local.example:3]: this file shouldn't be deleted. you can restore it with `git checkout master -- .env.local.example`

File: analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  onAtone(event: AtonementAnalyzerEvent) {
+    // const { penanceBoltNumber } = event.damageEvent as DirtyDamageEvent; Is this required?
+    let absorbedHealing = 0;
+    if ('sourceEvent' in event.healEvent && !('attackPower' in event.healEvent.sourceEvent)) {
+      // is a SpiritShellEvent
+      if ('absorbed' in event.healEvent.sourceEvent) {
+        absorbedHealing = event.healEvent.sourceEvent.absorbed || 0;
+      } else if (!('absorbed' in event.healEvent.sourceEvent)) {
```
Comment (emallson) [inline analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx]:
```suggestion
      } else {
```

File: analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        absorbedHealing = event.healEvent.sourceEvent.amount || 0;
+      }
+    } else if (!('sourceEvent' in event.healEvent)) {
```
Comment (emallson) [inline analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx]:
```suggestion
    } else {
```

**Review by emallson — CHANGES_REQUESTED** [review]
some small changes, but otherwise looks good

File: .env.local.example (hunk 1 above)
Comment (oisinod) [reply .env.local.example:3]: done

File: analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx (hunk 2 above)
Comment (oisinod) [reply analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx]: done

File: analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx (hunk 3 above)
Comment (oisinod) [reply analysis/priestdiscipline/src/modules/shadowlands/conduits/SwiftPenitence.tsx]: done

**Review by emallson — APPROVED** [review]

**Pull request merged**